src/main/java/com/puppycrawl/tools/checkstyle/checks/NewlineAtEndOfFileCheck.java

```diff
@@ -76,19 +76,8 @@
 
     @Override
     protected void processFiltered(File file, FileText fileText) {
-        // Cannot use lines as the line separators have been removed!
         try {
-            final RandomAccessFile randomAccessFile = new RandomAccessFile(file, "r");
-            boolean threw = true;
-            try {
-                if (!endsWithNewline(randomAccessFile)) {
-                    log(0, MSG_KEY_NO_NEWLINE_EOF, file.getPath());
-                }
-                threw = false;
-            }
-            finally {
-                Closeables.close(randomAccessFile, threw);
-            }
+            readAndCheckFile(file);
```

Comment:
Closables.close throws IOException and here the logic is build around threw variable. If we remove it and just put RandomAccessFile creation into try with resources - everything is ok, but than we broke the current logic.